src/instructlab/configuration.py

```diff
@@ -560,6 +609,49 @@ def init(ctx: click.Context, config_file: str | os.PathLike[str]) -> None:
     ctx.obj = Lab(config_obj, config_file, error_msg)
     if config_obj is not None:
         ctx.default_map = get_dict(config_obj)
+        # subtly get the expert args per cfg section
+        # if any are missing, add in sane defaults
+        train_additional = ctx.default_map["train"]["expert_args"]
+        ctx.default_map["train"]["expert_args"] = finish_expert_train_args(
+            train_additional
+        )
         log.configure_logging(log_level=config_obj.general.log_level.upper())
     else:
         ctx.default_map = None
+
+
+def map_train_to_library(params):
```

Comment:
@cdoern i kind of feel that while this indeed seperate out the args that are less likely for the user to change, the additional args are still very tightly coupled. This is because the `map_train_to_library` function needs to be maintained in the instructlab repo.. and every new argument that is added, this repo needs to be changed

I wonder if it is possible to `additional_args` schema-less, and pass to the `train` library to be mapped. Of course this requiers the current approach to be changed. Right now the issue is because we are forming `train_args` and `torch_args` to be passed into `run_training`

```python
        ds_args = DeepSpeedOptions(**params)
        lora_args = LoraOptions(**params)
        train_args = TrainingArgs(**params)
        torch_args = TorchrunArgs(**params)
        train_args.deepspeed_options = ds_args
        train_args.lora = lora_args
        try:
            run_training(train_args=train_args, torch_args=torch_args)
        except:
           ...
```

However, if we can somehow have only `TrainingArgs`, and just keep `addtional_args` schemaless, then make the call

```python
     train_args = TrainingArgs(**params)
     try:
            run_training(train_args=train_args, additional_args=additional_args)
      except:
           ...
```
and let the train library perform the mapping, we can minimize changes to the instructlab API. This will require the assumption that `addtional_args` are expert, since we are no longer going to expose the schema to the user.

@RobotSail @Maxusmusti @aldopareja any thoughts?

Reply:
> I wonder if it is possible to additional_args schema-less, and pass to the train library to be mapped. 

this scares me a bit. There needs to be some sort of accountability between the library and the cli in terms of breaking changes.

I went with this approach because it decreases the size of the cfg, makes it easier to understand, and allows us to be a bit less strict with the training library. 

I believe the mapping should probably happen in the CLI for accountability purposes but I am open to the training squad's opinions on this!

Reply:
well. even if you put a schema on the args, the `training` repo can still do some pretty bad things internally. I think maybe if the proper e2e tests are devised it will give more confidence

Reply:
@fabianlim While I don't disagree, this is something that's out of the scope of this PR, as it will be much bigger lift than what we can currently afford in terms of bandwidth.